packages/hardhat-core/test/internal/core/config/config-validation.ts

```diff
@@ -1812,4 +1815,25 @@ describe("Config validation", function () {
       });
     });
   });
+
+  describe("Resolved Config validation", function () {
```

Comment:
I think there should be several other test cases here:

- A similar test as the existing one, but iwth a valid number of runs
- One with `{ optimizer: enabled }` (no runs)
- One with the settings object but without the `optimizer` entry
- One where the invalid number of runs is in the overrides part of the config

There are other combinations, but that should be good enough.

Reply:
@fvictorio  
I have added all the remaining tests